polkadot/runtime/westend/src/lib.rs

```diff
@@ -596,15 +667,79 @@ impl pallet_bags_list::Config<VoterBagsListInstance> for Runtime {
 	type Score = sp_npos_elections::VoteWeight;
 }
 
-pallet_staking_reward_curve::build! {
-	const REWARD_CURVE: PiecewiseLinear<'static> = curve!(
-		min_inflation: 0_025_000,
-		max_inflation: 0_100_000,
-		ideal_stake: 0_500_000,
-		falloff: 0_050_000,
-		max_piece_count: 40,
-		test_precision: 0_005_000,
-	);
+// parameter_types! {
+// 	pub const NisBasePeriod: BlockNumber = 30 * DAYS;
+// 	pub MinReceipt: Perquintill = Perquintill::from_rational(1u64, 10_000_000u64);
+// 	pub const IntakePeriod: BlockNumber = 5 * MINUTES;
+// 	pub MaxIntakeWeight: Weight = MAXIMUM_BLOCK_WEIGHT / 10;
+// 	pub const ThawThrottle: (Perquintill, BlockNumber) = (Perquintill::from_percent(25), 5);
+// 	pub const NisPalletId: PalletId = PalletId(*b"py/nis  ");
+// }
+
+// impl pallet_nis::Config for Runtime {
+// 	type WeightInfo = weights::pallet_nis::WeightInfo<Runtime>;
+// 	type RuntimeEvent = RuntimeEvent;
+// 	type Currency = Balances;
+// 	type CurrencyBalance = Balance;
+// 	type FundOrigin = frame_system::EnsureSigned<AccountId>;
+// 	type Counterpart = NisCounterpartBalances;
+// 	type CounterpartAmount = WithMaximumOf<ConstU128<21_000_000_000_000_000_000u128>>;
+// 	type Deficit = (); // Mint
+// 	type IgnoredIssuance = ();
+// 	type Target = dynamic_params::nis::Target;
+// 	type PalletId = NisPalletId;
+// 	type QueueCount = ConstU32<300>;
+// 	type MaxQueueLen = ConstU32<1000>;
+// 	type FifoQueueLen = ConstU32<250>;
+// 	type BasePeriod = NisBasePeriod;
+// 	type MinBid = dynamic_params::nis::MinBid;
+// 	type MinReceipt = MinReceipt;
+// 	type IntakePeriod = IntakePeriod;
+// 	type MaxIntakeWeight = MaxIntakeWeight;
+// 	type ThawThrottle = ThawThrottle;
+// 	type RuntimeHoldReason = RuntimeHoldReason;
+// 	#[cfg(feature = "runtime-benchmarks")]
+// 	type BenchmarkSetup = ();
+// }
+
+pub struct EraPayout;
+impl pallet_staking::EraPayout<Balance> for EraPayout {
+	fn era_payout(
+		total_staked: Balance,
+		_total_issuance: Balance,
+		era_duration_millis: u64,
+	) -> (Balance, Balance) {
+		let auctioned_slots: u64 = 0;
+		// TODO: figure out if using parachains here makes sense
+		// // all para-ids that are currently active.
+		// let auctioned_slots = Paras::parachains()
+		// 	.into_iter()
+		// 	// all active para-ids that do not belong to a system chain is the number
+		// 	// of parachains that we should take into account for inflation.
+		// 	.filter(|i| *i >= LOWEST_PUBLIC_ID)
+		// 	.count() as u64;
+		const MAX_ANNUAL_INFLATION: Perquintill = Perquintill::from_percent(10);
+		const MILLISECONDS_PER_YEAR: u64 = 1000 * 3600 * 24 * 36525 / 100;
+
+		let use_auctioned_slots = dynamic_params::inflation::UseAuctionSlots::get();
+		let params = EraPayoutParams {
+			total_staked,
+			// TODO: update this to use NIS issuance
+			total_stakable: 1,
```

Comment:
Should this be a dynamic param instead of using the NIS pallet?

Reply:
No, this comes from the runtime, and depends on the specific runtime. 

For all runtimes that don't have NIS, this should simply be `balances::total_issuance()`. In NIS-enabled runtiems like Kusama, it comes from `NIS`.